### ISSUE TITLE
Problem consultation

### ISSUE THREAD

**Issue by MM666-code** [CityMind-Lab/ICML25-TimeVLM#5]
Thank you for your patient response to your previous question. But I have some questions when running your code. In the article, Vision Augmented Learner (VAL) uses Frequency and Periodicity Encoding, but there is no content about Frequency Encoding during the process of running class LearnableTimeseries ToImage (nn. Module):. Then I saw that the class MultiscaleLearnableTimestamperiesToImage (nn. Module) had Frequency Encoding, but it included Fast Fourier Transform and Wavelet Transform. When I changed the code to use class MultiscaleLearnableTimestamperiesToImage (nn. Module) as the tool for converting graphs, I found that the 0th dimension of the graph converted by this class was 244, while the 0th dimension of the subsequent text converted was 32, causing the code to run until multimodal_features = torch.cat([vision_embeddings, text_embeddings], dim=-1) # [B, hidden_size * 2] An error occurred because the 0th dimension does not match.
Question 1: Why does the class MultiscaleLearnableTimestamperiesToImage (nn. Module) have two time-frequency domains (Fast Fourier Transform and Wavelet Transform) that do not correspond to the original text?
Question 2 is: multimodal_features = torch.cat([vision_embeddings, text_embeddings], dim=-1) # [B, hidden_size * 2] An error occurred because the 0th dimension does not match. Can you guide me on how to modify the code?
    def forward_prediction(self, x_enc, vision_embeddings, text_embeddings):
        B, L, n_vars = x_enc.shape
        
        # 1. Process temporal features
        patches, _ = self.patch_embedding(x_enc.transpose(1, 2))  # [B * n_vars, n_patches, d_model]
        
        # 2. Compute local and global memory
        local_memory = self._compute_local_memory(patches)  # [B * n_vars, n_patches, d_model]
        global_memory = self._compute_global_memory(patches)  # [B * n_vars, 1, d_model]
        
        # 3. Combine local and global memory
        memory_features = local_memory + global_memory  # [B * n_vars, n_patches, d_model]

        # 4. Get temporal predictions
        memory_features = self.flatten(memory_features)  # [B * n_vars, head_nf]
        temporal_features = self.temporal_head(memory_features)  # [B, n_vars, d_model]
        memory_features = self.memory_head(memory_features)  # [B * n_vars, pred_len]
        temporal_features = einops.rearrange(temporal_features, '(b n) d -> b n d', b=B, n=n_vars)  # [B, n_vars, d_model]
        memory_features = einops.rearrange(memory_features, '(b n) d -> b n d', b=B, n=n_vars)  # [B, n_vars, pred_len]
        
        # 5. Process multimodal features
        multimodal_features = torch.cat([vision_embeddings, text_embeddings], dim=-1)  # [B, hidden_size * 2]
        multimodal_features = self.multimodal_enhancement(multimodal_features)  # [B, d_model]
        multimodal_features = multimodal_features.unsqueeze(1).expand(-1, n_vars, -1)  # [B, n_vars, d_model]
        multimodal_features = self.layer_norm(multimodal_features) 
We look forward to your reply.
